vendor/github.com/bitrise-io/go-steputils/v2/export/export.go

```diff
@@ -0,0 +1,177 @@
+package export
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/bitrise-io/go-utils/v2/command"
+	"github.com/bitrise-io/go-utils/v2/pathutil"
+	"github.com/bitrise-io/go-utils/ziputil"
+)
+
+const (
+	filesType              = "files"
+	foldersType            = "folders"
+	mixedFileAndFolderType = "mixed"
+)
+
+// Exporter ...
+type Exporter struct {
+	cmdFactory command.Factory
+}
+
+// NewExporter ...
+func NewExporter(cmdFactory command.Factory) Exporter {
+	return Exporter{cmdFactory: cmdFactory}
+}
+
+// ExportOutput is used for exposing values for other steps.
+// Regular env vars are isolated between steps, so instead of calling `os.Setenv()`, use this to explicitly expose
+// a value for subsequent steps.
+func (e *Exporter) ExportOutput(key, value string) error {
+	cmd := e.cmdFactory.Create("envman", []string{"add", "--key", key, "--value", value}, nil)
+	out, err := cmd.RunAndReturnTrimmedCombinedOutput()
+	if err != nil {
+		return fmt.Errorf("exporting output with envman failed: %s, output: %s", err, out)
+	}
+	return nil
+}
+
+// ExportOutputNoExpand works like ExportOutput but does not expand environment variables in the value.
+// This can be used when the value is unstrusted or is beyond the control of the step.
+func (e *Exporter) ExportOutputNoExpand(key, value string) error {
+	cmd := e.cmdFactory.Create("envman", []string{"add", "--key", key, "--value", value, "--no-expand"}, nil)
```

Comment:
Yes, I also think we should move to pure Go imports instead of calling the `envman` CLI. At that point, maybe steps could even directly import the `envman` package for the most basic usecases instead of the `export` package.

Reply:
Exposing all parameters: I originally wanted to add one more argument to the original function, but because Go doesn't have default argument values, this wouldn't work without breaking the function signature. The same reason why I want to avoid adding every possible envman flag, the call-side would become very awkward (`ExportOutput(key, value, nil, nil, nil, nil, false, nil, "")`

Reply:
What about this syntax (similarly how we handle command options in the v2 command package):
`ExportOutput(key, value, opts *Opts)`

We could use a new name for this function (`ExportOutputEnv`) and deprecate the original `ExportOutput`.